### PR TITLE
Docs for on-premise Docker

### PR DESCRIPTION
**Pull request by mattrobenolt**
This requires coordination with changes to `sentry-docs` or the build will break.

### CODE REVIEW

**Comment by dcramer** [comment]
- Let's not document "you can use these env vars" (dont remove support)
- Focus on MVP of "you have to compile your own image today"
- We need to document file store (as another ticket) since we seemingly dont today
- Let's figure out how to make SENTRY_FILESTORE_PATH more hardcoded/read-only

**Comment by mattrobenolt** [comment]
@getsentry/infrastructure I think this is ready for review now.

**Comment by mitsuhiko** [comment]
🍪

File: docs/installation/docker/index.rst [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+    make build
+
+If you plan on running the dependent services in Docker as well, we
+support linking containers.
+
+Running Dependent Services
+--------------------------
+
+*Running the dependent services in Docker is entirely optional*, but
```
Comment (JTCunning) [inline docs/installation/docker/index.rst]: Your choice here, but bolding this with ``**`` makes it definitely stand out more as opposed to italics.

File: docs/installation/docker/index.rst [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    docker run \
+      --detach \
+      --name sentry-smtp \
+      tianon/exim4
+
+Running Sentry Services
+-----------------------
+
+.. note:: The image that is built, acts as the entrypoint for all
+          running pieces for the Sentry application and the same image
```
Comment (JTCunning) [inline docs/installation/docker/index.rst]: This line and the following need to be aligned with `note::` to continue the sentence.

IE:
```
.. note:: The image that is built, acts as the entrypoint for all
   running pieces for the Sentry application and the same image
   must be used for all containers.
```

**Comment by JTCunning** [comment]
🎂

**Pull request merged**